src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -678,22 +678,35 @@ export class TerminalInstance implements ITerminalInstance {
 
 	public preparePathForTerminalAsync(path: string): Promise<string> {
 		return new Promise<string>(c => {
-			const hasSpace = path.indexOf(' ') !== -1;
-			if (platform.isWindows) {
-				const exe = this.shellLaunchConfig.executable;
-				if (!exe) {
+			const exe = this.shellLaunchConfig.executable;
+			if (!exe) {
+				if (platform.isWindows) {
 					c(path);
-					return;
+				} else {
+					c(this._escapeNonWindowsPath(path));
 				}
+				return;
+			}
+
+			const hasSpace = path.indexOf(' ') !== -1;
+			const isPowerShell = exe.indexOf('pwsh') !== -1 ||
```

Comment:
Should we be more explicit here? `path.basename('/bin/pwsh', '.exe')` and `path.basename('c:\\pwsh.exe')` should give `'pwsh'`

Reply:
If we want to be more explicit, we can use `path.basename` when we do: 

```ts
const exe = this.shellLaunchConfig.executable;
// becomes
const exe = path.basename(this.shellLaunchConfig.executable, '.exe')
```
Then we should be able to change all the `indexOf`s to `===` like the wsl ones:

```ts
if (((exe.indexOf('wsl') !== -1) || ((exe.indexOf('bash.exe') !== -1) && (exe.indexOf('git') === -1)))
```

Thoughts?

Reply:
fixed